nixos/modules/services/hardware/udev.nix

```diff
@@ -202,12 +202,26 @@ in
         '';
       };
 
-      extraRules = mkOption {
+      initrdRules = mkOption {
```

Comment:
Hmmph not particularily fond of having this option defined here.  This module is for configuring systemd-udevd in **stage 2**; not **stage 1** 

 how about instead of having an `initrdRules` here; we move this option to `stage-1.nix` and name it `boot.initrd.extraUdevRules` ?  It's a bit weird to have the the option for initrd defined here as `udevd` in initrd and `udevd` in stage-2 might be two completely different `udevd` implementations at this moment.

I'm working on an alternative initrd implementation using systemd. But it's hard to introduce this change as now the `udev` module in stage-2 has a hard dependency on the `stage-1.nix` module.  Because now  the `stage-2` udev module calls out to `stage-1.nix`  I can not disable `stage-1.nix` in `disabledModules`

Reply:
I'll send a draft PR with these suggested changes and i'll backlink it here.

Reply:
I think renaming this option is fine - as long as we do this before the 20.05 branchoff, it probably doesn't even need the `mkRenamedOption` alias.